fix(misc): a few fixes

daddykotex opened the pull request:
Here's a MR of most the fixes you made in your own fork. I had almost the same fixes after a few hours of testing.

Check it out and let me know if there is any issue.

The only thing I'm not sure about is the `sharedSession`. I could not get my test to work without it. The completion would never be called... Is it ok, I did not fully research this before changing it.

daddykotex commented:
I added a `package.json` so we can use `conventional-changelog` tool to generate the changelog: https://github.com/conventional-changelog/conventional-changelog-cli#recommended-workflow

paquetgp commented on Pod/Classes/MnuboClient.h:
```diff
@@ -29,6 +29,10 @@
+- (void)updateSmartObject:(MNUSmartObject *)smartObject withDeviceId:(NSString *)deviceId andCallBack:(void (^)(NSDictionary *data, NSError *error))callback;
+- (void)updateOwner:(MNUOwner *)owner withUsername:(NSString *)username andCallBack:(void (^)(NSDictionary *data, NSError *error))callback;
+- (void)sendEvents:(NSArray *)events withDeviceId:(NSString *)deviceId andCallBack:(void (^)(NSDictionary *data, NSError *error))callback;
 
```
Est-ce qu'il y a une raison pour laquelle on garderait les anciennes fonctions? backward compatibility?

daddykotex replied:
Yeah, the thing is that we have some users for the software. We could make them as deprecated, but no call back could still be a valid use case (fire & forget I guess).

paquetgp commented on Pod/Classes/MnuboClient.m:
```diff
@@ -63,19 +63,57 @@ - (BOOL)isOwnerConnected {
+
+// Services async
+
+- (void)updateSmartObject:(MNUSmartObject *)smartObject withDeviceId:(NSString *)deviceId andCallBack:(void (^)(NSDictionary *data, NSError *error))callback {
```
Pour le développement dans l'écosystème d'Apple, les callback sont généralement appelés des completion handler ou encore des block. Donc pour suivre le standard et le style du reste du code du SDK je garderais la syntaxe comme celle du login en utilisant la notation completion: completion.

daddykotex replied:
i'll update

paquetgp commented on Pod/Classes/MnuboClient.m:
```diff
@@ -63,19 +63,57 @@ - (BOOL)isOwnerConnected {
+    
+}
+
+- (NSArray*) convertEvents:(NSArray *)events {
```
Pour suivre le style de code, la signature devrait être comme ceci `- (NSArray *)convertEvents:(NSArray *)events`.

paquetgp commented on Pod/Classes/Services/MNUApiManager.m:
```diff
@@ -148,7 +148,7 @@ - (void)putWithPath:(NSString *)path body:(NSDictionary *)body completion:(void
         
         [MNUHTTPClient PUT:url headers:headers parameters:nil body:body completion:^(id data, NSDictionary *responsesHeaderFields, NSError *error) {
-            if (completion) completion(error);
+            if (completion) completion(nil, error);
```
Il devrait y avoir un check pour ajouter le data dans le `NSDictionnary *data` du completion si l'erreur est `nil`.

daddykotex replied:
The thing is that the body is not always a dictionnary, a.k.a. a json root object, it can be an array. In this case, the method calling the `putWithPath` is the only one aware of the returned values.

Here I completely discarded the returned values which is even worse, I'll update.

daddykotex commented:
I fixed the style issue, let me know what you think.

I also removed the data part from the completion handler of the few methods the client provides, because these API calls have no response body.
